importers/djornl/parser.py

```diff
@@ -709,13 +725,94 @@ def summarise_dataset(self):
                 'cluster': len(node_data['cluster']),
                 'full': len(node_data['full'])
             },
+            'errors_total': len(errs),
+            'errors': errs
         }
 
 
-if __name__ == '__main__':
+def format_summary(summary, output):
+    if output == 'json':
+        return json.dumps(summary)
+    node_type_counts = [count for count in summary['node_type_count'].values()]
+    edge_type_counts = [count for count in summary['node_type_count'].values()]
+    values = [
+        summary['nodes_total'],
+        summary['edges_total'],
+        summary['nodes_in_edge'],
+        summary['node_data_available']['key_only'],
+        summary['node_data_available']['cluster'],
+        summary['node_data_available']['full'],
+        summary.get('errors_total'),
+    ] + node_type_counts + edge_type_counts
+    value_width = max([len(str(value)) for value in values])
+    node_type_names = dict(__NO_TYPE__="No type")
+    node_types = "\n".join([(
+            f"{count:{value_width}} {node_type_names.get(ntype, ntype)}"
+            .format(value_width)
+        )
+        for ntype, count in summary['node_type_count'].items()
+    ])
+    edge_type_names = dict()
+    edge_types = "\n".join([(
+            f"{count:{value_width}} {edge_type_names.get(etype, etype)}"
+            .format(value_width)
+        )
+        for etype, count in summary['edge_type_count'].items()
+    ])
+    text_summary = f"""
+{summary['nodes_total']:{value_width}} Total nodes
+{summary['edges_total']:{value_width}} Total edges
+{summary['nodes_in_edge']:{value_width}} Nodes in edge
+---
+Node Types
+{node_types:{value_width}}
+---
+Edge Types
+{edge_types:{value_width}}
+---
+Node data available
+{summary['node_data_available']['key_only']:{value_width}} Key only
+{summary['node_data_available']['cluster']:{value_width}} Cluster
+{summary['node_data_available']['full']:{value_width}} Full
+---
+{summary.get('errors_total'):{value_width}} Errors
+""".format(value_width)
+    return text_summary
+
+
+def main():
+    argparser = argparse.ArgumentParser(description='Load DJORNL data')
+    argparser.add_argument(
+        '--debug', action='store_true',
+        help='Print errors in summary, by default only their count is printed.'
```

Comment:
I think it would make more sense to have this switch function in the opposite manner -- if the arg is supplied, only the error count is printed; if it isn't, the whole error list is printed. The aim is to maximise data quality and minimise errors, and having the whole list of errors printed out by default will increase the likelihood that those errors get fixed. A number is easy to ignore!

(I know that the current full data set has a ton of errors -- future iterations will, I hope, have far fewer!)